sycl/include/sycl/detail/string.hpp

```diff
@@ -0,0 +1,82 @@
+//==----------------- string.hpp - SYCL standard header file ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstring>
+
```

Comment:
Is it intentional that `string.hpp`/`string_view.hpp` are not self-contained, or in other words, can we `#include <string>` here?

Reply:
Sure. Thanks.